### PR TITLE
PingProtect Implementation

### PR DESCRIPTION
**Pull request by jeyanthanperiyasamy**
# JIRA Ticket

Please link jira ticket here

# Description

Briefly describe the change and any information that would help speedup the review and testing.

### CODE REVIEW

File: forgerock-auth/build.gradle [hunk 1]
```diff
@@ -93,7 +94,6 @@ dependencies {
 
     def coroutine_version = '1.7.2'
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:$coroutine_version"
-    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutine_version"
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/build.gradle:96]: we dont need this anymore ..all the coroutines are part of `org.jetbrains.kotlinx:kotlinx-coroutines-android:`

File: build.gradle [hunk 2]
```diff
@@ -9,23 +9,14 @@
 
 buildscript {
     ext {
-        kotlin_version = '1.8.0'
+        kotlin_version = '1.9.10'
         customCSSFile = projectDir.toString() + "/dokka/fr-backstage-styles.css"
         customLogoFile = projectDir.toString() + "/dokka/logo-icon.svg"
         customTemplatesFolder = file(projectDir.toString() + "/dokka/templates")
     }
-    repositories {
```
Comment (jeyanthanperiyasamy) [inline build.gradle:17]: moved all the repositories to settings.xml . @spetrov we may need to test the maven release to make sure this change works

File: gradle.properties [hunk 3]
```diff
@@ -25,6 +25,5 @@ android.useAndroidX=true
 GROUP=org.forgerock
 VERSION=4.2.0
 VERSION_CODE=18
-android.defaults.buildfeatures.buildconfig=true
```
Comment (jeyanthanperiyasamy) [inline gradle.properties:28]: this is deprecated , so moved this buildconfig to build.gradle

File: samples/app/src/main/java/com/example/app/env/EnvViewModel.kt [hunk 4]
```diff
@@ -143,22 +154,42 @@ class EnvViewModel : ViewModel() {
...
 
-    fun select(context: Context, options: FROptions) {
+    fun select(
+        context: Context,
+        options: FROptions,
+    ) {
         FRAuth.start(context, options)
         current = options
+
+        CoroutineScope(Dispatchers.Default).launch {
```
Comment (witrisna) [inline samples/app/src/main/java/com/example/app/env/EnvViewModel.kt:164]: May remove this from the sample, what about create another navigation menu to initialize the Protect SDK?

**Review by witrisna — APPROVED** [review]
Looks good to me, minor comment for the sample App.

**Review by spetrov — COMMENTED** [review]
Looks good to me! 👍🏻

File: samples/app/build.gradle [hunk 5]
```diff
@@ -49,27 +49,24 @@ android {
 
     buildFeatures {
         compose true
+        buildConfig true
```
Comment (rodrigoareis) [inline samples/app/build.gradle:52]: Update copyright

File: samples/app/src/main/java/com/example/app/env/EnvViewModel.kt [hunk 6]
```diff
@@ -12,124 +12,135 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.CoroutineScope
```
Comment (rodrigoareis) [inline samples/app/src/main/java/com/example/app/env/EnvViewModel.kt:15]: Update copyright

File: samples/app/src/main/java/com/example/app/journey/Journey.kt [hunk 7]
```diff
@@ -31,6 +31,8 @@ import com.example.app.callback.DeviceSigningVerifierCallback
 import com.example.app.callback.IdPCallback
 import com.example.app.callback.NameCallback
 import com.example.app.callback.PasswordCallback
+import com.example.app.callback.PingOneProtectEvaluationCallback
```
Comment (rodrigoareis) [inline samples/app/src/main/java/com/example/app/journey/Journey.kt:34]: Update copyright

File: samples/auth/build.gradle [hunk 8]
```diff
@@ -39,6 +39,10 @@ android {
         }
     }
 
+    buildFeatures {
```
Comment (rodrigoareis) [inline samples/auth/build.gradle:42]: Update copyright

File: samples/authenticator/build.gradle [hunk 9]
```diff
@@ -38,7 +38,8 @@ android {
         abortOnError false
     }
     buildFeatures {
-        viewBinding true
+        compose true
```
Comment (rodrigoareis) [inline samples/authenticator/build.gradle:41]: Update copyright

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Changes looks good. Minor comments to be addressed.

**Comment by jeyanthanperiyasamy** [comment]
comments addressed .

**Review by spetrov — APPROVED** [review]

**Review by witrisna — APPROVED** [review]

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Review by spetrov — CHANGES_REQUESTED** [review]

**Pull request merged**